main.py

```diff
@@ -336,6 +336,31 @@ def run_flask():
             debug=debug,
             threaded=True)
 
+### FILE READ FUNCTIONALITY
+from flask import flash, redirect, url_for
+from werkzeug.utils import secure_filename
+ALLOWED_EXTENSIONS = {'txt'}
+def allowed_file(filename):
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+sample_names_in_file = []
+@app.route('/', methods=['GET', 'POST'])
+def upload_file():
+    if request.method == 'POST':
+        content=''
+        if 'file' not in request.files:
+            content = 'No File Selected!'
+            return redirect(request.url)
+        file = request.files['file']
+        if file.filename == '':
+            content = 'Wrong Format File(use .txt)!'
+            return redirect(request.url)
+        if file and allowed_file(file.filename):
+            filename = secure_filename(file.filename)
+            content = file.read()
+        return render_template("index.html", text = content)
```

Comment:
I guess you see only the content of the file because of this line, no?

Reply:
Paolo is right. If you want to see other things too, you have to pass other variables into `render_template` function. At the moment you are trying to render index.html template which is the initial page of Samples. If you look at `index()` function, it's return statement in particular, you'll see that a bunch of other things should be passed too: campaigns, campaign_groups and pwgs. These are lists that are rendered in the page as tables. Campaigns are fetched from the sqlite database.
Note: after latest changes, I think only campaign_groups are necessary as initial page does not show separate campaigns anymore. This needs to be updated.

Reply:
Thanks a lot, it worked by changing the render_template call to include the other arguments from the index() function. I will try to find how to implement a search query for each line of the input text file to the database as the next step.